python/ray/serve/_private/application_state.py

```diff
@@ -416,6 +416,17 @@ def _set_target_state(
 
         self._target_state = target_state
 
+        # Log the resulting declarative + imperative version of the deployment config
+        # after each update to the application
+        if deployment_infos is not None and not deleting:
+            deployment_configs = {
+                name: info.to_dict() for name, info in deployment_infos.items()
+            }
+            logger.info(
+                f"Application '{self._name}' updated. Deployment configs (declarative + imperative): "
+                f"{json.dumps(deployment_configs, indent=2, default=str)}"
+            )
```

Comment:
i think the log in `deploy` is sufficient. this one is going to be too noisy since it get called on every autoscaling decision.